File: python-examples/computer-use-automation/hooks/setup_context.py
from intuned_runtime import attempt_store
from stagehand import Stagehand


async def setup_context(*, api_name: str, api_parameters: str, cdp_url: str):
    if api_name == "api/gemini-computer-use" or api_name == "api/stagehand-template":
        stagehand = Stagehand(
            env="LOCAL", local_browser_launch_options=dict(cdp_url=cdp_url)
        )
        await stagehand.init()
        attempt_store.set("stagehand", stagehand)

        async def cleanup():
            await stagehand.close()

        return stagehand.context, stagehand.page, cleanup

    elif api_name == "api/browser-use-template":
        import os
        os.environ["BROWSER_USE_CONFIG_DIR"] = "/tmp/browseruse-config"
        from browser_use import Browser

        browser = Browser(cdp_url=cdp_url)
        attempt_store.set("browser", browser)
        return

        
        

File: typescript-examples/computer-use/hooks/setupContext.ts
import { Stagehand, AISdkClient } from "@browserbasehq/stagehand";
import { createOpenAI } from "@ai-sdk/openai";
import { attemptStore, getAiGatewayConfig } from "@intuned/runtime";

async function getWebSocketUrl(cdpUrl: string) {
  console.log("Getting web socket URL from cdpUrl:", cdpUrl);
  // CDP endpoints need /json/version to get the webSocketDebuggerUrl
  const versionUrl = cdpUrl.endsWith('/') ? `${cdpUrl}json/version` : `${cdpUrl}/json/version`;
  const response = await fetch(versionUrl);
  const data = await response.json();
  return data.webSocketDebuggerUrl;
}

export default async function setupContext({
  cdpUrl,
  apiName,
  apiParameters,
}: {
  cdpUrl: string;
  apiName: string;
  apiParameters: any;
}) {
  if (apiName === "api/stagehand") {
    // Stagehand API uses AI gateway for Anthropic models
    const webSocketUrl = await getWebSocketUrl(cdpUrl);

    const { apiKey, baseUrl } = await getAiGatewayConfig();

    const openai = createOpenAI({
      apiKey,
      baseURL: baseUrl,
    });

    const llmClient = new AISdkClient({
      model: openai("gpt-4o"),
    });

    const stagehand = new Stagehand({
      env: "LOCAL",
      localBrowserLaunchOptions: {
        cdpUrl: webSocketUrl,
        viewport: {
          width: 1280,
          height: 800,
        },
      },
      llmClient,
      logger: console.log,
    });

    await stagehand.init();
    attemptStore.set("stagehand", stagehand);
  } else if (apiName === "api/gemini-computer-use") {
    // Gemini API uses environment variable for API key (AI gateway doesn't support Gemini yet)
    const webSocketUrl = await getWebSocketUrl(cdpUrl);

    const stagehand = new Stagehand({
      env: "LOCAL",
      localBrowserLaunchOptions: {
        cdpUrl: webSocketUrl,
        viewport: {
          width: 1280,
          height: 800,
        },
      },
      logger: console.log,
    });

    await stagehand.init();
    attemptStore.set("stagehand", stagehand);
  }
}
